packages/terra-table/src/Table.jsx

```diff
@@ -0,0 +1,352 @@
+};
+
+const defaultProps = {
+  rowHeaderIndex: 0,
+  defaultColumnWidth: 200,
+  columnHeaderHeight: '2.5rem',
+  rowHeight: '2.5rem',
+  pinnedColumns: [],
+  overflowColumns: [],
+  isScrollable: true,
+};
+
+// Default column size constraints
+const defaultColumnMinimumWidth = 60;
+const defaultColumnMaximumWidth = 300;
+
+function Table(props) {
+  const {
+    id,
+    ariaLabelledBy,
+    ariaLabel,
+    rows,
+    pinnedColumns,
+    overflowColumns,
+    onColumnResize,
+    defaultColumnWidth,
+    columnHeaderHeight,
+    rowHeight,
+    onColumnSelect,
+    onCellSelect,
+    rowHeaderIndex,
+    intl,
+  } = props;
+
+  const [pinnedColumnOffsets, setPinnedColumnOffsets] = useState([]);
+
+  // Initialize column width properties
+  const initializeColumn = (column) => ({
+    ...column,
+    width: column.width || defaultColumnWidth,
+    minimumWidth: column.minimumWidth || defaultColumnMinimumWidth,
+    maximumWidth: column.maximumWidth || defaultColumnMaximumWidth,
+  });
+
+  const displayedColumns = [
+    ...pinnedColumns,
+    ...overflowColumns,
+  ];
+  const [renderedColumns, setRenderedColumns] = useState(displayedColumns.map((column) => initializeColumn(column)));
+
+  // Manage column resize
+  const [tableHeight, setTableHeight] = useState(0);
+  const [activeIndex, setActiveIndex] = useState(null);
+  const [isContainerFocusable, setIsContainerFocusable] = useState(false);
+
+  const activeColumnPageX = useRef(0);
+  const activeColumnWidth = useRef(200);
+  const tableWidth = useRef(0);
+  const tableRef = useRef();
+  const tableContainerRef = useRef();
+  const sortedColumn = useRef();
+
+  const [cellAriaLiveMessage, setCellAriaLiveMessage] = useState(null);
+
+  const gridContext = useContext(GridContext);
+
+  // Define ColumnContext Provider value object
+  const columnContextValue = useMemo(() => ({ pinnedColumnOffsets, setCellAriaLiveMessage }), [pinnedColumnOffsets]);
+  const theme = useContext(ThemeContext);
+
+  // -------------------------------------
+  // functions
+
+  // -------------------------------------
+  // useEffect Hooks
+
+  // initialize the resize observer
+  useEffect(() => {
+    const resizeObserver = new ResizeObserver(() => {
+      // Update table height state variable
+      setTableHeight(tableRef.current.offsetHeight - 1);
+    });
+
+    // Register resize observer to detect size changes
+    resizeObserver.observe(tableRef.current);
+
+    return () => {
+      resizeObserver.disconnect();
+    };
+  }, [tableRef]);
+
+  // Make the table container focusable when scrollable.
+  useEffect(() => {
+    const containerRect = tableContainerRef.current.getBoundingClientRect();
+    const tableRect = tableRef.current.getBoundingClientRect();
+    if (tableRect.height > containerRect.height || tableRect.width > containerRect.width) {
+      setIsContainerFocusable(true);
+    }
+  }, [tableContainerRef, tableRef]);
```

Comment:
This does not seem correct.  We would probably want to use resize observers as the contents of the table can change.